src/traces/scattergeo/plot.js

```diff
@@ -116,19 +116,14 @@ function makeLineGeoJSON(trace) {
 }
 
 plotScatterGeo.plot = function(geo, scattergeoData) {
-    var gScatterGeo = geo.framework.select('g.scattergeolayer'),
-        topojson = geo.topojson;
-
-    // TODO move to more d3-idiomatic pattern (that's work on replot)
-    // N.B. html('') does not work in IE11
-    gScatterGeo.selectAll('*').remove();
-
-    var gScatterGeoTraces = gScatterGeo
-        .selectAll('g.trace.scatter')
+    var gScatterGeoTraces = geo.framework.select('.scattergeolayer')
+        .selectAll('g.trace.scattergeo')
         .data(scattergeoData);
 
     gScatterGeoTraces.enter().append('g')
-            .attr('class', 'trace scattergeo');
+        .attr('class', 'trace scattergeo');
+
+    gScatterGeoTraces.exit().remove();
 
     // TODO add hover - how?
```

Comment:
oh, we still need this unfortunately. scattergeo trace with mode `'lines'` don't have any hover text at the moment.